File: dashboard-core/src/main/java/com/dashboard/core/model/project/Project.java
package com.dashboard.core.model.project;

import com.dashboard.core.model.ticket.Ticket;
import com.dashboard.core.model.ticket.TicketSeverity;
import com.dashboard.core.model.user.UserGroup;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;
import java.util.List;

import static java.util.Collections.emptyList;

/**
 * @author Leboc Philippe.
 */
@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "projects")
public class Project {

    @Id
    @GeneratedValue
    private int id;

    @NotBlank
    private String label;

    @Size(max = 500)
    private String description;

    @OneToMany
    private List<Ticket> backlog;

    @OneToMany
    private List<Sprint> sprints;

    @OneToMany
    private List<TicketSeverity> ticketSeverities;

    @ManyToMany
    private List<UserGroup> userGroups;

    public Project(String label, String description) {
        this(-1, label, description, emptyList(), emptyList(), emptyList(), emptyList());
    }
}

File: dashboard-front/src/app/model/Project.ts
export class Project {
    public id:number;
    public label:string;
    public description:string;
    public backlog:any[];
    public sprints:any[];
    public ticketSeverities:any[];
    public userGroups:any[];


    constructor(id: number, label: string, description: string, backlog: any[], sprints: any[], ticketSeverities: any[], userGroups: any[]) {
        this.id = id;
        this.label = label;
        this.description = description;
        this.backlog = backlog;
        this.sprints = sprints;
        this.ticketSeverities = ticketSeverities;
        this.userGroups = userGroups;
    }
}
